Sources/Sentry/SentryProfiler.mm

```diff
@@ -143,7 +143,7 @@
             @"value" : obj[@"value"],
         }];
     }];
-    if (useMostRecentRecording && slicedGPUEntries.count == 0) {
+    if (useMostRecentRecording && slicedGPUEntries.count == 0 && nearestPredecessorValue != nil) {
```

Comment:
m: Does it make sense to have a test to avoid regression?

Reply:
I don't think so for several reasons: 
1. the only way this will regress is if someone removes the nil check
2. the testing is disproportionately difficult to implement
3. i don't even have a repro on this yet
4. i'm working on a more holistic solution in #3090

Reply:
+1 to fixing the immediate crash right now and working on the correct fix + tests in the other PR @armcknight linked